File: apps/ai-service/app/schemas/knowledge.py
from typing import List, Dict, Any, Optional
from pydantic import BaseModel, Field

class KnowledgeQueryRequest(BaseModel):
    query: str = Field(..., min_length=2, description="Natural language search query")
    org_id: str = Field(..., description="Authenticated organization ID")
    n_results: int = Field(default=3, ge=1, le=10, description="Number of results to return")
    include_demo: bool = Field(default=False, description="Whether to search demo collection (forbidden in production)")

class KnowledgeItem(BaseModel):
    id: str
    text: str
    metadata: Dict[str, Any]
    distance: Optional[float] = None

class KnowledgeQueryResponse(BaseModel):
    success: bool = True
    query: str
    org_id: str
    results: List[KnowledgeItem]

File: packages/shared-types/src/knowledge.ts
import { z } from 'zod';

export interface KnowledgeItemDto {
  id: string;
  text: string;
  metadata: {
    org_id: string;
    category?: string;
    title?: string;
    is_demo?: boolean;
    [key: string]: any;
  };
  distance?: number | null;
}

export interface KnowledgeQueryRequestDto {
  query: string;
  nResults?: number;
  includeDemo?: boolean;
}

export interface KnowledgeQueryResponseDto {
  success: boolean;
  query: string;
  org_id: string;
  results: KnowledgeItemDto[];
}

export const KnowledgeQueryRequestSchema = z.object({
  query: z.string().min(2),
  nResults: z.number().int().min(1).max(10).optional().default(3),
  includeDemo: z.boolean().optional().default(false)
});
